axum-extra/src/routing/mod.rs

```diff
@@ -159,22 +158,20 @@ pub trait RouterExt<B>: sealed::Sealed {
     ///     .route_with_tsr("/bar/", get(|| async {}));
     /// # let _: Router = app;
     /// ```
-    fn route_with_tsr<T>(self, path: &str, service: T) -> Self
+    fn route_with_tsr(self, path: &str, method_router: MethodRouter<S, B>) -> Self
```

Comment:
Also here (although I don't know off the top of my head why this exists twice in the first place).

Reply:
Its because its defined on an extension trait (`RouterExt`)